core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -264,6 +273,15 @@ public BlockOutStream getOutStream(long blockId, long blockSize, WorkerNetAddres
     }
     LOG.debug("Create block outstream for {} of block size {} at address {}, using options: {}",
         blockId, blockSize, address, options);
+
+    // ALLUXIO-11172: If the worker is in a container, use the container hostname
+    // to establish the connection.
+    if (!address.getContainerHost().equals("")) {
```

Comment:
this redundant translation can be avoided with the change in `NetworkAddressUtils`